Fix typos (#11911)
dotnet/docs: docs/csharp/delegates-overview.md

concept, and use that same concept across many different software
problems.

Second, the team wanted to support both single and multicast method

> calls. (Multicast delegates are delegates where multiple methods have

Reviewer: ```suggestion
calls. (Multicast delegates are delegates that chain together multiple method calls. 
```

> been chained together. You'll see examples

Reviewer: ```suggestion
You'll see examples
```
